### ISSUE TITLE
Undefined Error for GitLab

### ISSUE THREAD

**Issue by jamesmundia** [piceaTech/node-gitlab-2-github#20]
Getting undefined error:

```
undefined
/home/jamesmundia/dev/node-gitlab-2-github/index.js:20
if(settings.gitlab.url === "http://gitlab.mycompany.com/"){
           ^

TypeError: Cannot read property 'gitlab' of undefined
    at Object.<anonymous> (/home/jamesmundia/dev/node-gitlab-2-github/index.js:20:12)
    at Module._compile (module.js:410:26)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Function.Module.runMain (module.js:442:10)
    at startup (node.js:136:18)
    at node.js:966:3
```

gitlab portion of settings.json file:

```
  "gitlab": {
    "url": "https://gitlab.com/MYUSERNAME/MYREPONAME",
    "token": "{{TOKEN}}",
    "projectID": ID NUMBER
  },
```

Tried many variations but can't get past the undefined message

**Comment by markustenghamn** [comment]
I received the same error when I had an extra comma in my json file. Use a json validator to make sure you don't have any errors . My error was in the username mappings.

**Comment by spruce** [comment]
@jamesmundia Have you found a fix for that?

**Comment by jamesmundia** [comment]
@markustenghamn thanks for the tip - I'll try that out when I'm at the console later. @spruce no fix yet, but will update after testing tonight

**Comment by jamesmundia** [comment]
Still not seeing success, even after running it through a validator. Anybody want to post a successfull settings file, I can see where I'm messing up?

**Comment by spruce** [comment]
Will try it and report back with an example settings file

**Comment by spruce** [comment]
```
{
  "gitlab": {
    "url": "https://gitlab.com/",
    "token": "My Token",
    "projectID": 3161232
  },
  "github": {
    "url": "api.github.com",
    "pathPrefix": "",
    "owner": "piceaTech",
    "username": "spruce",
    "password": "Password",
    "repo": "testProject"
  },
  "usermap": {
    "Spruce": "spruce"
  },
  "projectmap": {
    "Spruce/TestProjekt": "piceaTech/testProject"
  }
}
```

you can see the original here: https://gitlab.com/Spruce/TestProjekt and the copy here: https://github.com/piceaTech/testProject

**Comment by jamesmundia** [comment]
Finally got it running! 

In the sample settings, URL is beneath token for GitLab. I reversed these on my successful settings file. 

I also had to run a curl command to get the correct project ID from GitLab, found on SO: http://stackoverflow.com/questions/39559689/where-do-i-find-the-project-id-for-the-gitlab-api

(I had not done this before to get the proper projectID)

There we some extra commas for me, even after these edits, and after those were removed it worked.
